### PR TITLE
Feature eslint plugin filenames

### PR DESCRIPTION
**Pull request by chaseadamsio**
Adds plugin filenames to walmartlabs best-practices and updates docs to reflect change. cc @ryan-roemer

### CODE REVIEW

**Comment by ryan-roemer** [comment]
@chaseadamsio @baer -- Does this work with the React configuration? I think I remember Eric saying that the `_.merge()` can't handle 2+ plugins. (Hence one of the reasons behind me filing #13 ...)

**Comment by ryan-roemer** [comment]
Yep, confirmed that the use of `lodash` *does* produce a bug:

```
rye@titan ~/scm/wm/eslint-config-defaults $ git diff
diff --git a/configurations/walmart/es6-react.js b/configurations/walmart/es6-react.js
index fae7810..bc03fb1 100644
--- a/configurations/walmart/es6-react.js
+++ b/configurations/walmart/es6-react.js
@@ -14,3 +14,5 @@ module.exports = _.merge({},
     }
   }
 );
+
+console.log("es6-react", module.exports);
rye@titan ~/scm/wm/eslint-config-defaults $ node configurations/walmart/es6-react.js 
es6-react { env: { es6: true, node: false },
  ecmaFeatures: { modules: true, jsx: false },
  globals: { require: false, module: false, process: false, fetch: false },
  rules: 
   { /* SNIPPED */ },
  plugins: [ 'react' ],
  parser: 'babel-eslint' }
rye@titan ~/scm/wm/eslint-config-defaults $ 
```

I think it's high time we did #13 and removed lodash, and instead relied on eslint to merge configs like it is specifically built to do.

**Comment by chaseadamsio** [comment]
Cool. I'll move on to #13 and come back to this next. :+1:

**Comment by baer** [comment]
#13 is a bit blocked. I did not find the exposed _merge function @ryan-roemer was referring to and need to investigate more.

**Comment by chaseadamsio** [comment]
I think #13 is all good if #23 looks okay, I tested and everything is :+1:

**Comment by baer** [comment]
You forgot the off.js config

**Comment by chaseadamsio** [comment]
#13 is closed and merged, I've made some changes to fit the schema @baer was setting up with the `react` plugin, so this should close #17, was there anything else that should be done here @ryan-roemer?

File: README.md [hunk 1]
```diff
@@ -22,25 +22,29 @@ $ npm install --save-dev eslint-config-defaults
...
-**This package is made for for ESLint 1.0.0+ and eslint-plugin-react 3.1.0+ and [eslint-config-airbnb](https://www.npmjs.com/package/eslint-config-airbnb) 0.0.7**
+**This package is made for for ESLint 0.24.0+**
 
 ### Full Configurations
 
 This package includes the following complete and ready to use configurations:
 
+_**Note:**_ Any React config `<prefix>-react` requires [eslint-plugin-react](https://www.npmjs.com/package/eslint-plugin-react)
+
+_**Note:**_ Any Walmart config `defaults/configurations/walmart/<suffix>` requires [eslint-plugin-filenames](https://github.com/selaux/eslint-plugin-filenames)
```
Comment (ryan-roemer) [inline README.md]: Excellent.

**Comment by ryan-roemer** [comment]
@chaseadamsio -- LGTM!

File: README.md (hunk 1 above)
Comment (chaseadamsio) [inline README.md]: That part was copy/pasta from the upstanding Mr. @baer.

File: README.md [hunk 2]
```diff
@@ -22,25 +22,29 @@ $ npm install --save-dev eslint-config-defaults
 
 ## Usage
 
-**This package is made for for ESLint 1.0.0+ and eslint-plugin-react 3.1.0+ and [eslint-config-airbnb](https://www.npmjs.com/package/eslint-config-airbnb) 0.0.7**
+**This package is made for for ESLint 0.24.0+**
```
Comment (baer) [inline README.md]: You probably didn't intend to delete this.

**Comment by baer** [comment]
This is so :moneybag:!

**Pull request merged**